velox/exec/tests/utils/LocalExchangeSource.cpp

```diff
@@ -29,26 +29,33 @@ class LocalExchangeSource : public exec::ExchangeSource {
       memory::MemoryPool* pool)
       : ExchangeSource(taskId, destination, queue, pool) {}
 
+  bool supportsFlowControlV2() const override {
+    return true;
+  }
+
   bool shouldRequestLocked() override {
     if (atEnd_) {
       return false;
     }
     return !requestPending_.exchange(true);
   }
 
-  ContinueFuture request(uint32_t maxBytes) override {
+  folly::SemiFuture<Response> request(
+      uint32_t maxBytes,
+      uint32_t maxWaitSeconds) override {
     ++numRequests_;
 
-    auto [promise, future] =
-        makeVeloxContinuePromiseContract("LocalExchangeSource::request");
+    auto promise = VeloxPromise<Response>("LocalExchangeSource::request");
+    auto future = promise.getSemiFuture();
+
     if (numRequests_ % 2 == 0) {
```

Comment:
What is this logic? Why every second request we return 0?
Should we add a comment explaining it?

Reply:
This is a test source, so to make things a bit more interesting I added to logic to simulate "no data" response every other time. There is a comment: "// Simulate no-data." This is pre-existing change.

```
    if (numRequests_ % 2 == 0) {
      {
        std::lock_guard<std::mutex> l(queue_->mutex());
        requestPending_ = false;
      }
      // Simulate no-data.
      promise.setValue(Response{0, false});
      return future;
    }
```